codecov/CMakeLists.txt

```diff
@@ -0,0 +1,115 @@
+# CMake targets for generating code coverage reports
+#
+# Note that the targets in here are not needed to enable code coverage
+# on builds. To have builds generate code coverage metadata, one only
+# has to enable the --coverage option for the compiler and this is
+# done in the top-level CMakeLists.txt so that the option covers all
+# build targets in the project.
+#
+# Given that all dependencies (lcov, genhtml) are installed, and CMake
+# is initialized with -DCODECOVERAGE=ON, it should be
+# possible to generate a code coverage report by running:
+#
+# $ make coverage
+#
+# The report is generated in REPORT_DIR and can be viewed in a web
+# browser.
+
+# Find lcov for html output
+find_program(LCOV lcov)
+
+if (LCOV)
```

Comment:
Just a thought, maybe instead of putting everything into large if block just do something like this instead to reduce nesting:

``` cmake
if (NOT LCONV)
  message(STATUS "Install lcov to generate code coverage reports")
  return()
endif()
```